pipelines/utils/temporal_coverage_updater/utils.py

```diff
@@ -0,0 +1,289 @@
+                        {query_class}({_input}){{
+                        edges{{
+                            node{{
+                            startYear,
+                            startMonth,
+                            startDay,
+                            interval,
+                            }}
+                        }}
+                        }}
+                    }}"""
+    r = requests.post(
+        url="https://api.basedosdados.org/api/v1/graphql",
+        json={"query": query, "variables": dict(zip(keys, values))},
+        headers=header,
+    ).json()
+    return r
+
+
+def create_update(
+    mutation_class,
+    mutation_parameters,
+    query_class,
+    query_parameters,
+    update=False,
+):
+    EMAIL = os.environ["API_EMAIL"]
+    PASSWORD = os.environ["API_PASSWORD"]
+    backend = b.Backend(graphql_url="http://api.basedosdados.org/api/v1/graphql")
+    token = get_token(backend, email=EMAIL, password=PASSWORD)
+    header = {
+        "Authorization": f"Bearer {token}",
+    }
+
+    r, id = get_id(
+        query_class=query_class, query_parameters=query_parameters, header=header
+    )
+    if id is not None:
+        r["r"] = "query"
+        if update is False:
+            return r, id
+
+    _classe = mutation_class.replace("CreateUpdate", "").lower()
+    query = f"""
+                mutation($input:{mutation_class}Input!){{
+                    {mutation_class}(input: $input){{
+                    errors {{
+                        field,
+                        messages
+                    }},
+                    clientMutationId,
+                    {_classe} {{
+                        id,
+                    }}
+                }}
+                }}
+            """
+
+    if update is True and id is not None:
+        mutation_parameters["id"] = id
+    r = requests.post(
+        "https://api.basedosdados.org/api/v1/graphql",
+        json={"query": query, "variables": {"input": mutation_parameters}},
+        headers=header,
+    ).json()
+
+    r["r"] = "mutation"
+    if "data" in r and r is not None:
+        if r.get("data", {}).get(mutation_class, {}).get("errors", []) != []:
+            print(f"create: not found {mutation_class}", mutation_parameters)
+            print("create: error\n", json.dumps(r, indent=4, ensure_ascii=False), "\n")
+            id = None
+            raise Exception("create: Error")
+        else:
+            id = r["data"][mutation_class][_classe]["id"]
+            # print(f"create: created {id}")
+            id = id.split(":")[1]
+
+            return r, id
+    else:
+        print("\n", "create: query\n", query, "\n")
+        print(
+            "create: input\n",
+            json.dumps(mutation_parameters, indent=4, ensure_ascii=False),
+            "\n",
+        )
+        print("create: error\n", json.dumps(r, indent=4, ensure_ascii=False), "\n")
+        raise Exception("create: Error")
+
+
+def parse_temporal_coverage(temporal_coverage):
```

Comment:
essa função ta funcionando? o formato de entrada que vc passa parece ser `{first_date}{last_date}` e parece que a função está montada pro nosso antigo modelo de metadados que é `{first_date}(frequency_interval){last_date}` 

Talvez valha a pena comentar melhor pra gente conseguir entender melhor o que tá acontecendo, ou talvez simplificar se não for necessário todos esses if's

Reply:
@laura-l-amaral  funciona sim, mas ainda preciso documentar (ela já existia no repo).
Ela foi feita p/ o modelo antigo msm, mas a variável `first_date` eu tô imprimindo já com o intervalo, então funciona. 

Por exemplo: para o `test_dataset.test_laura_student` o first date retornado pela função `get_first_date()` é: `1900-1-1(1)` ao invés de só `1900-1-1`. 

Ai quando roda o `parse_temporal_coverage(f"{first_date}{last_date}")` ele retorna o dicionário direitinho tipo:  

```
{'startYear': 1900,
 'startMonth': 1,
 'startDay': 1,
 'endYear': 2023,
 'endMonth': 5,
 'endDay': 24,
 'interval': 1}
```